test/wasm/oom.js

```diff
@@ -0,0 +1,40 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+function assertEquals(expected, actual) {
+    if (expected != actual) {
+        throw `Expected ${expected}, received ${actual}`;
+    }
+}
+
+function wasmAlloc(initialSize, newSize) {
+
+    let memories = [];
+    const n = 5;
+
+    for (let i = 0; i < n; i++) {
+        try {
+            let m = new WebAssembly.Memory({initial:initialSize});
+            assertEquals(initialSize * (1 << 16) /*64K*/, m.buffer.byteLength);
+            m.grow(newSize);
+            memories.push(m);
+        } catch (e) {
+            return 0;
+        }
+    }
+
+    return 1;
+}
+
+assertEquals(2, WScript.Arguments.length);
+
+const INITIAL_SIZE = parseInt(WScript.Arguments[0]);
+const GROW_SIZE = parseInt(WScript.Arguments[1]);
+
+assertEquals(0, wasmAlloc(INITIAL_SIZE, GROW_SIZE));
```

Comment:
@Cellule 
Spec seems to suggest that `Memory.prototype.grow` is supposed to throw in case of OOM

> Let ret be the result of performing a grow_memory operation given delta d.
> If ret is -1, a RangeError is thrown.

https://github.com/WebAssembly/design/blob/master/JS.md#webassemblymemoryprototypegrow

I could add a test that calls `grow_memory` directly. However, it looks like testing `Memory.prototype.grow`  is sufficient since the only two other possible exceptions `Memory.prototype.grow` could throw are `WASMERR_NeedMemoryObject` and `JSERR_DetachedTypedArray` are unlikely to be thrown in this particular scenario. I could add an extra test to check the error's message to make sure the correct error (`JSERR_ArgumentOutOfRange` ) is thrown if you think this is necessary. 

Could you please elaborate on this one? 

> Plus you are not checking that we correctly return -1 when we get an out of memory exception (in c++)